src/lib.rs

```diff
@@ -16,11 +25,77 @@
 use std::convert::From;
 use std::io::{Result, Write};
 use std::mem;
+use std::ops;
+use std::fmt;
 
 /// A digest.
-pub type Digest = [u8; 16];
+///
+/// The raw data of the digest can be accessed with the `raw_data()` method.
+/// You can also obtain a hexadecimal representation (which is usually used
+/// to display MD5-hashes) by using the hex formatter from `std::fmt`:
+///
+/// ```
+/// let digest = md5::compute(b"Ferris");
+/// println!("{:x}", digest); // prints 'f87eb669657376982ea0184c3c27c0bc'
+/// println!("{:X}", digest); // prints 'F87EB669657376982EA0184C3C27C0BC'
+///
+/// // The hexadecimal representation can also be saved in a `String` via the
+/// // `format!()` macro.
+/// let hex_string = format!("{:x}", digest); // : String
+/// assert_eq!(hex_string, "f87eb669657376982ea0184c3c27c0bc");
+/// ```
+///
+pub struct Digest(pub [u8; 16]);
+
+impl Digest {
+    /// Returns the raw data as fixed-size array. Of course, you can also
+    /// just access the field via `digest.0`, but this method is more
+    /// descriptive.
+    pub fn raw_data(&self) -> [u8; 16] {
```

Comment:
I think you are talking about `Into`? Because an impl like `impl From<Digest> for [u8; 16]` is impossible thanks to orphan rules.

I'm not sure about that. The `Into` impl founds fine, but a custom method can be handy, too. Although I'm not very happy about the name `raw_data`. `as_array` or something like that would be possible as well...